src/OrchardCore.Modules/OrchardCore.Users/Views/UserFields.Edit.cshtml

```diff
@@ -40,20 +43,81 @@
         </div>
 
         <div class="mt-2">
-            <div class="btn btn-warning btn-sm" id="generatePassword">@T["Generate password"]</div>
+            <div class="btn btn-warning btn-sm" id="generatePassword" data-password-length="@Math.Max(IdentityOptions.Value.Password.RequiredLength, IdentityOptions.Value.Password.RequiredUniqueChars)">@T["Generate password"]</div>
             <div class="btn btn-info btn-sm" id="copyPassword">@T["Copy password"]</div>
         </div>
     </div>
-
     <script at="Foot" asp-name="userFieldsPassword">
         document.addEventListener('DOMContentLoaded', function() {
-            const strongPasswordLength = 12;
             let password = document.getElementById('@Html.IdFor(p => p.Password)');
             let passwordConfirmation = document.getElementById('@Html.IdFor(p => p.PasswordConfirmation)');
             let passwordToggle = document.getElementById('passwordToggle');
             let passwordConfirmationToggle = document.getElementById('passwordConfirmationToggle');
             let generate = document.getElementById('generatePassword');
             let copy = document.getElementById('copyPassword');
+            const requiredPasswordLength = parseInt(generate.getAttribute('data-password-length'));
+            const allowed = {
+                uppers: 'ABCDEFGHIJKLMNOPQRSTUVWXYZ',
+                lowers: 'abcdefghijklmnopqrstuvwxyz',
+                digits: '0123456789',
+                symbols: '!#$%&\()*+,-./:;<=>\'?@@[\]^_`{|}~'
+            };
+
+            function getRandomChar(str) {
+                return str.charAt(Math.floor(Math.random() * str.length));
+            }
+            function shuffle(str) {
```

Comment:
Add new line between functions

Reply:
This is inline code, the idea to to reduce the amount of code injected in the HTML output. you sure you want me to add extra line between functions?

Reply:
I got what you mean, you can check how we did it in other modules

Reply:
I do not see consistence in OC when it comes to scripts. In some cases I see new line and other not. I don't really care, if you want me to make another commit and add extra spaces I will.

Reply:
is this better?

Reply:
> I don't really care

But we care ;) hope to make things better in the upcoming future. Don't forgot we built the framework with limited resources and help the community, but we are struggling to improve things as much as we can

Reply:
By I don’t care I mean, I don’t care either way. Obviously I care about the success of this framework which is why I continue to contribute.